streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -183,6 +183,73 @@ abstract class ZStream[-R, +E, +O](
       } yield pull
     }
 
+  private final def bufferSignal[E1 >: E, O1 >: O](
```

Comment:
Sorry, @regiskuckaertz, but how does this code deal with resuming after error? I believe you have removed the mechanism for that. Doesn't this just stop after it encounters the first error?

Reply:
I was looking at `upstream` and it cannot fail from the inside, whereas in downstream, the `pull` intercepts any error and set the flag `done` to true. After that, `pull` will always yield `Pull.End`. Is my reasoning correct? I'll add safety tests to make sure of it! But this was a test to figure out why bufferSliding is behaving differently, and it seems that downstream is pulling the end signal which blows up the test with an error of `None` as it should. But I wonder why it is pulling that end signal in the first place 🤔

Reply:
I'm talking about the `Promise` for each element that paused the pipeline until the error has been handled downstream.

Reply:
Ha, yes that's the thing I missed. Thanks : 🙏